### PR TITLE
Improve Dropdown Interaction

### PR DESCRIPTION
**Pull request by puneetjohal**
Improved dropdown functionality by adding the following features:

- The dropdown stays expanded when a user selects a filter or clicks on the search box to manually search for a filter
- The dropdown stays expanded as users are deselecting filters
- The search box clears the text after the user selects a filter from the dropdown to prepare for a new search
- The dropdown displays a message saying "No results found." if the search does not match any applicable filters, or if there are no more filters to choose from
- The arrow/triangle on the right points in different directions depending on whether the dropdown is expanded or not
- The dropdown will appear disabled if there are no pre-selected filters and there are no filters available to choose from (visually, the dropdown will be grayed out, no special cursors will appear when hovering over different sections of the dropdown, the placeholder text will read "No filters apply", and the dropdown will be unable to expand)

### CODE REVIEW

File: frontend/css/_DropdownFilter.scss [hunk 1]
```diff
@@ -88,6 +112,19 @@
     width: 100%;
   }
 
+  &__noResults {
```
Comment (gamburgm) [inline frontend/css/_DropdownFilter.scss]: can you rename this to `&--noResults`? Take a look at the [BEM spec](http://getbem.com/naming/), will hopefully clarify what I mean here.

**Review by gamburgm — CHANGES_REQUESTED** [review]
One small change, and then I think we're good to ship!

**Review by NEUDitao — APPROVED** [review]

File: frontend/css/_DropdownFilter.scss (hunk 1 above)
Comment (puneetjohal) [reply frontend/css/_DropdownFilter.scss]: Done (in both the stylesheet and DropdownFilter.tsx)

**Review by gamburgm — APPROVED** [review]
Yeet!

**Pull request merged**